python/dynamodb-lambda/lambda/consumer/lambda_function.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import print_function
+
+import json
+import decimal
+import os
+import boto3
+from botocore.exceptions import ClientError
+
+
+# Helper class to convert a DynamoDB item to JSON.
+class DecimalEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, decimal.Decimal):
+            if o % 1 > 0:
+                return float(o)
+            else:
+                return int(o)
+        return super(DecimalEncoder, self).default(o)
+
+
+# Get the service resource.
+dynamodb = boto3.resource('dynamodb')
+
+# set environment variable
+TABLE_NAME = os.environ['TABLE_NAME']
+
+
+def lambda_handler(event, context):
+    table = dynamodb.Table(TABLE_NAME)
+    # Scan items in table
+    try:
+        response = table.scan()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `scan` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `ExclusiveStartKey`, `IsTruncated`, `LastEvaluatedKey`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)